File: react-native-totvs-clockin-vision/android/src/main/java/com/totvs/clockin/vision/face/VisionFaceCamera.kt
package com.totvs.clockin.vision.face

import android.graphics.Bitmap
import com.totvs.clockin.vision.core.RecognitionModel
import com.totvs.clockin.vision.core.VisionCamera
import java.io.File

/**
 * [VisionCamera] dedicated to face detection/recognition
 */
interface VisionFaceCamera : VisionCamera {

    /**
     * Configure a liveness strategy to use in this vision camera.
     */
    var liveness: Liveness?

    /**
     * Face proximity feature
     */
    var proximity: Proximity?

    /**
     * Whether or not this face vision camera has a liveness feature installed
     */
    val isLivenessEnabled: Boolean

    /**
     * Whether or not this face vision camera has a proximity feature installed
     */
    val isFaceProximityEnabled: Boolean

    /**
     * Setup this [VisionFaceCamera] with a proper model and options
     */
    fun setup(model: RecognitionModel<Bitmap, Face>)

    /**
     * Capture and perform recognition task on this face vision camera view.
     *
     * @param options control certain aspects of the recognition process.
     * @param onResult callback called once the recognition task is completed.
     */
    fun recognizeStillPicture(options: RecognitionOptions, onResult: (RecognitionResult) -> Unit)

    /**
     * Options to control aspects of the recognition on still pictures.
     *
     * @param saveImage whether we need to save the captured image or not.
     * @param outputDir location where to save the image.
     */
    data class RecognitionOptions(val saveImage: Boolean = false, val outputDir: File? = null)

    /**
     * Result of a recognition tasks.
     *
     * @param file in which the image was saved.
     * @param faces recognized from the image.
     */
    data class RecognitionResult(var file: File? = null, var faces: List<Face> = emptyList())
}

File: react-native/react-native-totvs-clockin-vision/ios/Face/VisionFaceCamera.swift
//
//  VisionFaceCameraswift.swift
//  TOTVSClockInVision
//
//  Created by Jansel Rodriguez on 7/30/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

/**
* [VisionCamera] dedicated to face detection/recognition
*/
public protocol VisionFaceCamera : VisionCamera {
    /**
     * RGB to be used as color for overlay graphics on this camera view.
     */
    var overlayGraphicsColor: String { get set }
    
    /**
     * Configure a liveness strategy to use in this vision camera.
     */
    var liveness: Liveness? { get set }
    
    /**
     * Face proximity feature
     */
    var proximity: Proximity? { get set }
    
    /**
     * Whether or not this face vision camera has a liveness feature installed
     */
    var  isLivenessEnabled: Bool { get }
    
    /**
     * Whether or not this face vision camera has a proximity feature installed
     */
    var isFaceProximityEnabled: Bool { get }
    
    /**
     * Setup this [VisionFaceCamera] with a proper model and options
     */
    func setup(model: RecognitionDetectionModel<String, Face>)
    
    /**
     * Capture and perform recognition task on this face vision camera view.
     *
     * @param options control certain aspects of the recognition process.
     * @param onResult callback called once the recognition task is completed.
     */
    func recognizeStillPicture(options: RecognitionOptions, onResult: (RecognitionResult) -> Void)
}
